### DISCUSSION TITLE
[QUESTION] Use CSI driver (or PVC) as longhorn disk

### DISCUSSION THREAD

**Discussion by AdamJSoftware** [longhorn/longhorn#6897] [General]
## Question

I am wondering if it is possible to use the k8s CSI driver to provision a disk for longhorn. In my specific case. I want to use longhorn on LKE. The problem is, that Linode likes to complete delete nodes used on LKE which includes their disk data. The alternative is to attach a block storage device to your node and use that for permanent storage. There are two ways of doing that, automatically via the CSI driver, or manually by attaching the volume to the node. That's what I have been doing up till this point. However, it becomes a hassle as Linode does not like to remount the volume after the node has been recycled (this is the term for creating a deleting nodes in the K8S cluster). They will only automatically attach via the CSI driver.

To summarize my question, is it possible to create a longhorn disk from a Persistent Volume?

**Comment by DamiaSan** [comment]
Maybe you can have a look at https://github.com/longhorn/longhorn/discussions/2190

**Comment by sreisich** [comment]
I'm experiencing a similar issue with my managed Kubernetes cluster from IONOS. The nodes are ephemeral and completely replaced each week, and their disks are purged. This also occurs whenever a configuration change is made to the node pool.

I'm wondering if it's possible to use Persistent Volume Claims from a IONOS storage class, such as "ionos-enterprise-ssd", as the base disks for Longhorn. Does anyone have insights or suggestions on how to achieve this?

It would be ideal if I could configure Longhorn during installation to avoid using the ephemeral node disks and instead create new PVCs with the storage class "ionos-enterprise-ssd" to use as disks. Any guidance on how to set this up, or something similar, would be greatly appreciated.

**Comment by RashKash103** [comment]
@AdamJSoftware I've found myself in the same predicament. Have you since been able to find a way to do this? If not, were you able to find a workaround to allow you to use Longhorn on LKE?

Thanks!
